cranelift/filetests/filetests/isa/aarch64/condbr.clif

```diff
@@ -2,18 +2,18 @@ test compile precise-output
 set unwind_info=false
 target aarch64
 
-function %f(i64, i64) -> b1 {
+function %f(i64, i64) -> i8 {
 block0(v0: i64, v1: i64):
   v2 = icmp eq v0, v1
   return v2
 }
 
 ; block0:
 ;   subs xzr, x0, x1
-;   cset x0, eq
+;   csetm x0, eq
```

Comment:
Is this right? I would expect an `icmp` to give us an `i8` with value either `0` or `1`, but here we're actually getting `0` or `-1` I think...

Reply:
This was due to the implementation of `materialize_bool_result` in the aarch64 lowerings special-casing a result of `1` bit to produce values of `1`/`0`, and otherwise producing `-1`/`0`. I've removed the special case, and it now unconditionally produces `1`/`0`. The downside of that change is that it complicates the lowering of `bmask`, but I think that's a reasonable trade-off.

Reply:
Cool, makes sense, thanks!

Do we have any runtests that return the result of an `icmp` or friends and directly assert its value? Seems like if not, that would be good coverage: value is 0 or 1 for various cases for each kind of comparison (`icmp`, `fcmp`).

Reply:
We have a lot of tests that perform a comparison, `uextend` it to a larger type, and then test the resulting values. The `runtests/sqrt.clif` test is a good example of this.

Reply:
Awesome, we're covered then, thanks! (I guess I should just review the tests in general; still need to do that, sorry!)

Reply:
It might be worth refactoring those tests to return the value of the `icmp`/`fcmp` directly, as there might be interference from the `uextend`.